### ISSUE TITLE
aonames in NWChem/Psi/QChem

### ISSUE THREAD

**Issue by ATenderholt** [cclib/cclib#158]
It looks like `aonames` is missing from the NWChem parser. Is there a keyword that would print an appropriate section? Right now, the only part of the log files that appears useful is:

```
 Summary of "ao basis" -> "ao basis" (cartesian)
 ------------------------------------------------------------------------------
       Tag                 Description            Shells   Functions and Types
 ---------------- ------------------------------  ------  ---------------------
 C                           sto-3g                  3        5   2s1p
 H                           sto-3g                  1        1   1s
```

It looks like the `aonames` would have to be generated manually by parsing the `Functions and Types` and using `atomnos`.

**Comment by berquist** [comment]
This is what I've found (taken from [here](http://www.nwchem-sw.org/index.php/Special:AWCforum/st/id1279/large_basis_set_and_printing_MO.html)):

> There is no easy way to get all the eigenvectors printed.
> You would need to edit the source code.
> For the DFT case, you would need to edit
> `$NWCHEM_TOP/src/nwdft/scf_dft/dft_scf.F`
> and change the line
>      `parameter (eval_pr_tol_lo = -1.5d0, eval_pr_tol_hi=0.5)`
> to
>      `parameter (eval_pr_tol_lo = -1d10, eval_pr_tol_hi=1d10)`

Even then, the ordering in the eigvenvector printout is by absolute magnitude of the coefficients. We would need to look at the source code.

**Comment by ATenderholt** [comment]
What's the section in ```dvb_sp_ks.out``` with ```Final MO vectors```? That looks like eigenvectors to me.

**Comment by berquist** [comment]
Yes, they are. If you look starting at line 413 of `C_bigbasis.out`:

```
 Vector   11  Occ=0.000000D+00  E= 2.190632D-01
              MO Center= -5.2D-15,  2.6D-14, -3.1D-14, r^2=-2.6D+00
   Bfn.  Coefficient  Atom+Function         Bfn.  Coefficient  Atom+Function
  ----- ------------  ---------------      ----- ------------  ---------------
    69      1.093908  1 C  dzz               67     -1.032529  1 C  dyy
    51     -0.903962  1 C  dzz               49      0.853255  1 C  dyy
    65     -0.503697  1 C  dxy              139      0.489583  1 C  gzzzz
   128     -0.479620  1 C  gxxyy            135     -0.461294  1 C  gyyyy
   130      0.448506  1 C  gxxzz             45     -0.444394  1 C  dzz
```
The functions aren't printed in order of increasing angular momentum, but decreasing MO coefficient. Though, perhaps they could be reordered by looking at the `Bfn.` column.

**Comment by ATenderholt** [comment]
I think ```C_bigbasis.out``` needs to be re-run with ```print "final vectors" "final vectors analysis"``` so that there is a section like the following:

```
                                 Final MO vectors
                                 ----------------


 global array: alpha evecs[1:60,1:60],  handle: -995 

            1           2           3           4           5           6  
       ----------- ----------- ----------- ----------- ----------- -----------
   1      -0.69930    -0.69930    -0.02746    -0.02769    -0.00313    -0.02871
   2      -0.03156    -0.03135     0.00410     0.00406     0.00078     0.00816
   3       0.00002    -0.00003     0.00067     0.00065    -0.00526    -0.00120
   4       0.00091     0.00063     0.00416     0.00413     0.00055    -0.00309
   5       0.00000    -0.00000    -0.00000     0.00000     0.00000    -0.00000
```

**Comment by ATenderholt** [comment]
I just committed [d1d259a5cd](https://github.com/ATenderholt/cclib/commit/d1d259a5cdee206e3cb45987c95958b1f3df3803) with the C_bigbasis run with spherical basis functions and the print statements.

**Comment by ATenderholt** [comment]
Ok, I think I have ```aonames``` correctly generated for NWChem, at least for the ```dvb_*``` and ```C_bigbasis``` examples---I see it breaks parsing the water MP2 and CCSD(t) examples because those have labels for the hydrogens.

I also have not put in the h-orbital labels.

**Comment by ATenderholt** [comment]
It might be hard to handle the edge cases for atom labels (from the NWChem docs):

"If the entry for <tag> begins with either the symbol or name of an element (regardless of case), then the center is treated as an atom of that type. The default charge is the atomic number (adjusted for the presence of ECPs by the ECP NELEC directive ; see Section 8). Additional characters can be added to the string, to distinguish between atoms of the same element (For example, the tags oxygen, O, o34, olonepair, and Oxygen-ether, will all be interpreted as oxygen atoms.)."

**Comment by langner** [comment]
This has been around for so long that we should probably close as obsolete. If we're concerned about the issue, but do not anticipate a fix anytime soon, let's adds comments to the code/tests to this effect with maybe some TODOs.

Will review before v1.5.3 gets out.

@ATenderholt thoughts?

**Comment by berquist** [comment]
* Psi4: according to the docs, it isn't implemented.
* Q-Chem: this is the current output:
```
Attempting to read data/QChem/basicQChem4.2/dvb_sp.out
aonames
['C1_1S', 'C1_2S', 'C1_2PX', 'C1_2PY', 'C1_2PZ', 'C2_1S',
'C2_2S', 'C2_2PX', 'C2_2PY', 'C2_2PZ', 'C3_1S', 'C3_2S',
'C3_2PX', 'C3_2PY', 'C3_2PZ', 'C4_1S', 'C4_2S', 'C4_2PX',
'C4_2PY', 'C4_2PZ', 'C5_1S', 'C5_2S', 'C5_2PX', 'C5_2PY',
'C5_2PZ', 'C6_1S', 'C6_2S', 'C6_2PX', 'C6_2PY', 'C6_2PZ',
'H7_1S', 'H8_1S', 'H9_1S', 'H10_1S', 'C11_1S', 'C11_2S',
'C11_2PX', 'C11_2PY', 'C11_2PZ', 'C12_1S', 'C12_2S',
'C12_2PX', 'C12_2PY', 'C12_2PZ', 'H13_1S', 'H14_1S',
'H15_1S', 'C16_1S', 'C16_2S', 'C16_2PX', 'C16_2PY',
'C16_2PZ', 'C17_1S', 'C17_2S', 'C17_2PX', 'C17_2PY',
'C17_2PZ', 'H18_1S', 'H19_1S', 'H20_1S']
```
which looks to be consistent with the Gaussian example in the docs.
